Model/Sync/Catalog/Processor/Main.php

```diff
@@ -404,7 +404,7 @@ protected function getApiParams(
 
                 if (!$this->getImmediateRetryAlreadyDone(
                     $this->entity,
-                    (int)$productId,
+                    $visibleVariants.(int)$productId,
```

Comment:
Can you please explain this change?

In case _visibleVariants_ is true, the value here will be `1{productId}`. Why is this needed?

Reply:
If a product sync has resulted in a 404/409, we should retry it once again. To avoid multiple retries for the same product, we are flagging whether a product is retried or not. $visibleVariants flag will be true if a product is a variant and at the same time it is an individually visible product. So these types of products will be part of both variant API and products API. To uniquely flag the retry attempts of these type of products , we are using this $visibleVariants value.